### PR TITLE
set and get esx subnets

### PR DESCRIPTION
**Pull request by promiseowolabi**
# Description
Setting and retrieving the ESXi subnets to be used.
`set_esx_subnets`
`get_esx_subnets`

Please describe your pull request in detail.

## Related Issue

This project only accepts pull requests related to open issues.

https://github.com/rubrikinc/rubrik-sdk-for-python/issues/175

## Motivation and Context

Why is this change required? What problem does it solve?

## How Has This Been Tested?

* Please describe in detail how you tested your changes.
* Include details of your testing environment, and the tests you ran to see how your change affects other areas of the code, etc.

Tested locally on gaia.

## Screenshots (if appropriate):
```
[2019-12-18 13:06:41,095] [DEBUG] -- set_esx_subnets: Setting the subnets that should be used to reach ESXi hosts: 'None'.
[2019-12-18 13:06:41,095] [DEBUG] -- PATCH https://emea1-rbk01.rubrikdemo.com/api/internal/vmware/config/set_esx_subnets
[2019-12-18 13:06:41,095] [DEBUG] -- Config: {"esxSubnets": ""}
[2019-12-18 13:06:42,681] [DEBUG] -- <Response [200]>

[2019-12-18 13:06:42,681] [DEBUG] -- get_esx_subnets: Retrieving the preferred subnets used to reach the ESXi hosts.
[2019-12-18 13:06:42,681] [DEBUG] -- GET https://emea1-rbk01.rubrikdemo.com/api/internal/vmware/config/esx_subnets
[2019-12-18 13:06:43,905] [DEBUG] -- <Response [200]>

{'esxSubnets': ''}
```
## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

Go over all the following points, and put an `x` in all the boxes that apply. If you're unsure about any of these, don't hesitate to ask. We're here to help!
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **[CONTRIBUTION](https://github.com/rubrikinc/rubrik-sdk-for-python/blob/master/CONTRIBUTING.md)** document.
- [ ] I have updated the CHANGELOG file accordingly for the version that this merge modifies.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: rubrik_cdm/data_management.py [hunk 1]
```diff
@@ -1995,3 +1995,36 @@ def sql_db_export(self, db_name, date, time, sql_instance=None, sql_host=None, t
                     target_database_name))
 
             return self.post('v1', '/mssql/db/{}/export'.format(mssql_id), config, timeout)
+
+    def set_esx_subnets(self, esx_subnets=None, timeout=15):  # pylint: ignore
```
Comment (drew-russell) [inline rubrik_cdm/data_management.py]: `set_esx_subnets`

Need to add an impotence check to this function

File: docs/set_esx_subnets.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+Sets the subnets that should be used to reach the ESXi hosts.
+```py
+def set_esx_subnets(esx_subnets=None, timeout=15)
+```
+
+## Keyword Arguments
+| Name        | Type | Description                                                                 | Choices | Default |
+|-------------|------|-----------------------------------------------------------------------------|---------|---------|
+| esx_subnets | str  | Preferred subnets used to reach the ESX hosts. The format should be the comma separated list without any spaces, for example, 192.168.2.10/24,10.255.0.2/16.  |         |         |
```
Comment (drew-russell) [inline docs/set_esx_subnets.md]: `The format should be the comma separated list without any spaces, for example, 192.168.2.10/24,10.255.0.2/16. `

Why not just make this a list instead?

Comment (promiseowolabi) [reply docs/set_esx_subnets.md]: The API request is expecting a string, are you suggesting taking a list and then concatenating the items into a string for the request? that will work too.

Comment (drew-russell) [reply docs/set_esx_subnets.md]: Yeah, let’s automatically convert to a string based off of the provided list.

File: docs/set_esxi_subnets.md [hunk 3]
```diff
@@ -1,14 +1,14 @@
...
 ```py
-def set_esx_subnets(esx_subnets=None, timeout=15)
+def set_esxi_subnets(esx_subnets=None, timeout=15)
 ```
 
 ## Keyword Arguments
 | Name        | Type | Description                                                                 | Choices | Default |
 |-------------|------|-----------------------------------------------------------------------------|---------|---------|
-| esx_subnets | str  | Preferred subnets used to reach the ESX hosts. The format should be the comma separated list without any spaces, for example, 192.168.2.10/24,10.255.0.2/16.  |         |         |
+| esx_subnets | list | Preferred subnets used to reach the ESX hosts. The format should be a list, for example ['192.168.2.10/24','10.255.0.2/16'].  |         |         |
```
Comment (drew-russell) [inline docs/set_esxi_subnets.md]: `The format should be a list, for example ['192.168.2.10/24','10.255.0.2/16']`

nit: We already define it as a list so we can just provide the example here.

File: rubrik_cdm/data_management.py [hunk 4]
```diff
@@ -1996,28 +1996,34 @@ def sql_db_export(self, db_name, date, time, sql_instance=None, sql_host=None, t
...
             timeout {int} -- The number of seconds to wait to establish a connection with the Rubrik cluster before returning a timeout error. (default: {15})
         Returns:
             dict -- The full response of `PATCH /internal/vmware/config/set_esx_subnets`.
         """
+        self.log("set_esx_subnets: Getting the existing subnets used to reach the ESXi hosts")
+        subnets_data = self.get_esxi_subnets()
+        subnets = subnets_data['esxSubnets'].split(',')
 
         config = {}
         if esx_subnets is None:
```
Comment (drew-russell) [inline rubrik_cdm/data_management.py]: Let's also add a check to validate it's a list and return an appropriate error message if its not

Comment (promiseowolabi) [reply rubrik_cdm/data_management.py]: Good call, made the changes now

File: docs/set_esxi_subnets.md (hunk 3 above)
Comment (promiseowolabi) [reply docs/set_esxi_subnets.md]: 👍

**Pull request merged**